### DISCUSSION TITLE
Result classes - new object proposal

### DISCUSSION THREAD

**Discussion by MartinSalak** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#235] [Ideas]
We would like to create a new object which could combine specific load cases and load combinations for results

New object: **StructuralResultClass**
Result classes can combine various load classes and combination together to create an envelope which can be later used for checks and viewing results. They are used to exclude some combinations from results or view results caused by specific load cases or combinations no .

| Property | Example | Description |
| --- | --- | --- |
| Name | RC1 | Unique name of the class |
| Description | All ultimate limit state combinations | Description of the class |
| Load cases | LC1, LC4 | List of all the load cases inside the class, reference to sheet StructuralLoadCase |
| Load combinations | COM1, COM3 | List of all the load combinations inside the class, reference to sheet StructuralLoadCombination |
| Id | | Unique attribute designation |

Notes:
Class can’t be empty. It must contain at least one load case or load combination.

**Comment by robinblazek** [comment]
Hi all, 
based on the remark from @NextFEM (thanks for the input!) we extend the proposal by lists of multipliers, which could be used for load case or combination. This solution is more flexible and it should cover even more use cases. 

Property | Example | Description
-- | -- | --
Name | RC1 | Unique name of the class
Description | All ultimate limit state combinations | Description of the class
Load case multipliers | 1.35; 1.5 | List of multipliers for load cases, first multiplier applies to first load case, etc. Count of multipliers must correspond to count of the load cases 
Load cases | LC1, LC4 | List of all the load cases inside the class, reference to sheet StructuralLoadCase
Load combi multipliers | 1.2; 1.4 | List of multipliers for load combination, first multiplier applies to first load combination, etc. Count of multipliers must correspond to count of the load combinations 
Load combinations | COM1, COM3 | List of all the load combinations inside the class, reference to sheet StructuralLoadCombination
Id |   | Unique attribute designation  

Let us know what you think about it!
